code/game/objects/items/storage/fancy.dm

```diff
@@ -248,12 +255,18 @@
 		quick_remove_item(/obj/item/clothing/mask/cigarette, user)
 	return CLICK_ACTION_SUCCESS
 
-/// Removes an item from the packet if there is one
-/obj/item/storage/fancy/cigarettes/proc/quick_remove_item(obj/item/grabbies, mob/user)
+/// Removes an item or puts it in mouth from the packet, if any
+/obj/item/storage/fancy/cigarettes/proc/quick_remove_item(obj/item/grabbies, mob/user, equip = 0)
 	var/obj/item/finger = locate(grabbies) in contents
 	if(finger)
-		atom_storage.remove_single(user, finger, drop_location())
-		user.put_in_hands(finger)
+		if(!equip)
+			atom_storage.remove_single(user, finger, drop_location())
+			user.put_in_hands(finger)
+			return
```

Comment:
`equip` should probably be renamed to something like `into_hands` or `equip_to_mouth`.

Reply:
👍